### PR TITLE
Closing the close graph prompt exits graph without save

### PR DESCRIPTION
**Pull request by Auriga2**
### Description of the Change
When closing an unsaved graph, closing the 'Close' dialog no longer opens the 'Save Graph' dialog. 
When cancelling or closing the 'Save Graph' dialog, the unsaved graph is no longer discarded.

### Alternate Designs
N/A

### Why Should This Be In Core?
Better UX as this is preventing users from loosing unsaved graphs.

### Benefits
Better UX as this is preventing users from loosing unsaved graphs.

### Possible Drawbacks
N/A

### Verification Process
Case A

Open new graph
Make a change
X on graph tab to close
X on close prompt
-> No Save Graph Dialog opening and close operation is cancelled (unsaved graph still exists).

Case B
Open new graph
Make a change
X on graph tab to close
Save
Close Save File Dialog || Press Cancel
->Close operation is cancelled (unsaved graph still exists)

### Applicable Issues
#508

### CODE REVIEW

**Comment by aldebaran30701** [comment]
I know this wasn't in the original ticket, but I'm wondering if you'd like to check out if the following can be fixed. When you have an unsaved graph, and close constellation with the X, it prompts the save menu. When you click X (or cancel) on that save menu - it consequently closes Constellation and discards the graph.
If possible, it should cancel the close event - to match the above implementation.

**Comment by Auriga2** [comment]
Hmm it's hard to debug, but nice if we can fix this too.

File: CoreInteractiveGraph/src/au/gov/asd/tac/constellation/graph/interaction/plugins/io/SaveAsAction.java [hunk 1]
```diff
@@ -181,10 +182,14 @@ public boolean isEnabled() {
...
 
+    public boolean isSaved() {
+        return isSaved;
+    } 
+
     @Override
     public void actionPerformed(final ActionEvent e) {
         refreshListeners();
-        Collection<? extends SaveAsCapable> inst = lkpInfo.allInstances();
+        Collection<? extends SaveAsCapable> inst = lkpInfo.allInstances();
```
Comment (cygnus-x-1) [inline CoreInteractiveGraph/src/au/gov/asd/tac/constellation/graph/interaction/plugins/io/SaveAsAction.java:192]: Looks like you've added some random whitespace to the end of this line. It's not a big enough deal to worry about fixing now, but there is an 'on save' option in Netbeans which will prevent this sort of thing next time.

Comment (Auriga2) [reply CoreInteractiveGraph/src/au/gov/asd/tac/constellation/graph/interaction/plugins/io/SaveAsAction.java]: That's a good tip. Thanks. Removed this.

**Comment by arcturus2** [comment]
@aldebaran30701 would you be ok if we merge this PR in and @Auriga2 can look at your suggestion when possible?

**Comment by aldebaran30701** [comment]
Yep I'm fine with this

**Pull request merged**